eodatasets3/wagl.py

```diff
@@ -264,29 +473,39 @@ def _create_contiguity(
     """
     for product in product_list:
         contiguity = None
+        # A contiguity layer for each product
         for grid, band_name, path in p.iter_measurement_paths():
             if not band_name.startswith(f"{product.lower()}:"):
                 continue
             # Only our given res group (no pan band in Landsat)
             if grid.resolution_yx != resolution_yx:
                 continue
-
             with rasterio.open(path) as ds:
                 ds: DatasetReader
-                if contiguity is None:
-                    contiguity = numpy.ones((ds.height, ds.width), dtype="uint8")
-                    geobox = GridSpec.from_rio(ds)
-                elif ds.shape != contiguity.shape:
-                    raise NotImplementedError(
-                        "Contiguity from measurements of different shape"
-                    )
-
-                for band in ds.indexes:
-                    contiguity &= ds.read(band) > 0
+                out_shape = ds.shape
```

Comment:
If I'm reading this correctly, this isn't guaranteed to be the correct resolution, as we're iterating _all_ measurements, not just the higher resolution one?

Reply:
Ah, nevermind. It's skipping the non-res ones above.